internal/core/docker_host.go

```diff
@@ -262,6 +278,17 @@ func dockerHostFromContext(ctx context.Context) (string, error) {
 	return "", ErrDockerSocketNotSetInContext
 }
 
+// dockerHostFromDockerContext returns the docker host from the DOCKER_CONTEXT environment variable, if it's not empty
+func dockerHostFromDockerContext(_ context.Context) (string, error) {
+	// exec `docker context inspect -f='{{.Endpoints.docker.Host}}'`
+	cmd := exec.Command("docker", "context", "inspect", "-f", "{{.Endpoints.docker.Host}}")
```

Comment:
suggestion: we have [an ongoing PR](https://github.com/testcontainers/testcontainers-go/pull/2810) addressing the docker context support. Although shelling out to the docker CLI is easy, we disregard its usage to avoid handling exit codes, and parsing responses.